File: Math, Logical Implememt/GCD, LCM/acmicpc.net/17087/Main.js
"use strict";

const inputFilePath = "/sample.txt";
const input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  .toString()
  .trim()
  .split("\n");
// const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const [n, s] = input[0].split(" ").map((x) => Number(x));
const siblings = input[1].split(" ").map((x) => Number(x));
const dis = [];

const GDC = (a, b) => {
  while (b !== 0) {
    const tmpA = a;
    const tmpB = b;
    a = tmpB;
    b = tmpA % tmpB;
  }
  return a;
};

siblings.forEach((sibling) => {
  dis.push(Math.abs(s - sibling));
});
let ans = dis[0];
for (let i = 1; i < dis.length; i++) {
  ans = GDC(ans, dis[i]);
}
console.log(ans);

File: Math, Logical Implememt/GCD, LCM/acmicpc.net/17087/Main.py
# import sys
# input = sys.stdin.readline
file = open("./01 ing/acmicpc.net/sample.txt", "r")
input = file.readline
n, s = map(int, input().split())
siblings = list(map(int, input().split()))


def GCD(a, b):
    while(b != 0):
        tmpA = a
        tmpB = b
        a = tmpB
        b = tmpA % tmpB
    return a


distance = []
for i in range(len(siblings)):
    distance.append(abs(siblings[i]-s))
ans = distance[0]
for i in range(1, len(distance)):
    ans = GCD(ans, distance[i])
print(ans)
